src/controllers/expenses.js

```diff
@@ -0,0 +1,118 @@
+'use strict';
+
+const { getAllUsers } = require('./users');
+
+let expenses = [];
+
+const getAll = (req, res) => {
+  const { userId, categories, from, to } = req.query;
+
+  if (userId) {
+    expenses = expenses.filter(expense => expense.userId === +userId);
+  }
+
+  if (categories) {
+    expenses = expenses.filter(({ category }) => (
+      categories.includes(category)
+    ));
+  }
+
+  if (from && to) {
+    expenses = expenses.filter(expense => {
+      const expenseDate = new Date(expense.spentAt);
+      const fromDate = new Date(from);
+      const toDate = new Date(to);
```

Comment:
You should parse the dates before filtering. Parsing dates inside the filter function is inefficient.
You create it each iteration